pylint/checkers/base.py

```diff
@@ -1719,7 +1719,31 @@ def _create_naming_options():
     return tuple(name_options)
 
 
-class NameChecker(_BasicChecker):
+class NameCheckerHelper:
+    """Class containing functions required by NameChecker and NonAsciiNamesChecker"""
+
+    def _check_name(
+        self, node_type: str, name: str, node: nodes.NodeNG, optional_kwarg: Any = None
+    ):
+        """Only Dummy function will be overwritten by implementing classes
+
+        Note: kwarg arguments will be different in implementing classes
+        """
+        raise NotImplementedError
+
+    def _recursive_check_names(self, args: Iterable[nodes.AssignName]):
+        """check names in a possibly recursive list <arg>"""
+        for arg in args:
+            if isinstance(arg, nodes.AssignName):
+                self._check_name("argument", arg.name, arg)
```

Comment:
This is always true right?

Reply:
Again I did not write this code I just moved it, to be able to use it.
I won't change anything there, because I don't understand the impacts. This should be an Issue for people that have written the code/understand it correctly.

Reply:
I'll do a follow-up after this has been merged!